scripts/fsutil.sh

```diff
@@ -0,0 +1,320 @@
+#!/bin/bash
+## This script is to create the tmpfs for different mount point during 
+## the ONIE installer image generation to speed up the processing.
+##
+## USAGE:
+##   ./tmpfs <storage> <operation>
+## ENVIRONMENT:
+##   storage  => Can be imagefs, rootfs, overlayfs, dockerfs, installerfs or allfs
+##   type     => It can be one of the operation: allocate, deallocate, create, delete, mount or umount
+
+## Include common functions
+. functions.sh
+
+## Enable debug output for script
+set -x 
+
+
+## Use tmpfs
+if [[ "${SONIC_USE_TMPFS_BUILD}" == "y" ]]; then
+	BUILD_TMP_ROOTFS_SIZE=6G
+	BUILD_ROOTFS_FILE_SIZE=4 #4G
+	BUILD_TMP_DOCKERFS_SIZE=16G
+	BUILD_TMP_FS_SIZE=8G
+else
+	BUILD_ROOTFS_FILE_SIZE=10 #10G
+fi
+
+## Read ONIE image related config file
+. ./onie-image.conf
+
+## Working directory to prepare the file system
+FILESYSTEM_BASE=/sonic/build/${BASE_IMAGE_NAME}
+FILESYSTEM_ROOT=${FILESYSTEM_BASE}/
+mkdir -p ${FILESYSTEM_ROOT}
+
+
+FSROOT_IMAGE_BASE_DIR=target/image/
+FSROOT_IMAGE_DIR=${FSROOT_IMAGE_BASE_DIR}/${BASE_IMAGE_NAME}
+mkdir -p ${FSROOT_IMAGE_DIR}
+FSROOT_IMAGE_FILENAME=fsroot.img
+FSROOT_IMAGE_FILE=${FSROOT_IMAGE_DIR}/${FSROOT_IMAGE_FILENAME}
+BUILD_PATH=target/fs/${TARGET_MACHINE}
+
+
+mount_imagefs()
+{
+	umount_allfs
+	delete_rootfs_image
```

Comment:
Looks likes mount_imagefs does some additional cleanup options? Can we remove them? Better to have another command to do the cleanup explicitly.